addons/mail/static/src/component_hooks/use_messaging/use_messaging.js

```diff
@@ -0,0 +1,11 @@
+/** @odoo-module **/
+
+const { useComponent } = owl.hooks;
+
+export async function useMessaging() {
+    const component = useComponent();
+    if (!component.env.isMessagingInitialized()) {
+        await component.env.messagingCreatedPromise;
+        component.render();
```

Comment:
This is strange, `isMessagingInitialized` becomes true when `init_messaging` flow is fully done, while `messagingCreatedPromise` is resolved when the core code of model is deployed, which is well before `isMessagingInitialized` is set.

Also from the look of the code, it seems this hook is used for top-level components that have `useModels()`. Would be best to combine this behavior for all components that have `useModels()` so that we just keep a single hook `useModels()`.

Reply:
Waiting on `messagingCreatedPromise` is enough here because the goal is to avoid crash of `this.env.messaging.` being `undefined` when reading any field.

The component might or might not make conditions on messaging being actually initialized depending on what it wants to do. I guess the condition should simply be on messaging created then.

I'll check to include it into the other hook instead.